packages/@aws-cdk/core/lib/token.ts

```diff
@@ -112,8 +112,9 @@ export class Tokenization {
    */
   public static resolve(obj: any, options: ResolveOptions): any {
     return resolve(obj, {
-      ...options,
-      preparing: false
+      scope: options.scope,
+      resolver: options.resolver,
+      preparing: options.preparing || false
```

Comment:
`||` is problematic for boolean fields (albeit here it will work). Prefer `options.preparing !== undefined ? options.preparing : false`.